release: 2.71 (#15733)
canonical/snapd: NEWS.md

# New in snapd 2.71
* FDE: auto-repair when recovery key is used
* FDE: revoke keys on shim update
* FDE: revoke old TPM keys when dbx has been updated
* FDE: do not reseal FDE hook keys every time

> * FDE: store keys in keyring in install in initrd

Reviewer: Is this a typo? "in keyring in install in initrd"?

Author: fixed

* FDE: allow disabled DMA on Core
* FDE: snap-bootstrap: do not check for partition in scan-disk on CVM

> * FDE: support secboot preinstall check for 25.10+ hybrid installs via the /v2/system/{label} endpoint
> * FDE: support generating recovery key at install time via the /v2/systems/{system-label} endpoint
> * FDE: update passphrase quality check at install time via the /v2/systems/{system-label} endpoint

Reviewer: For consistency, these should probably all be the same

```suggestion
* FDE: support secboot preinstall check for 25.10+ hybrid installs via the /v2/system/{label} endpoint
* FDE: support generating recovery key at install time via the /v2/systems/{label} endpoint
* FDE: update passphrase quality check at install time via the /v2/systems/{label} endpoint
```

Author: fixed

* FDE: support replacing recovery key at runtime via the new /v2/system-volumes endpoint

> * FDE: support checking recovery keys add runtime via the /v2/system-volumes endpoint

Author: fixed

* FDE: support enumerating keyslots at runtime via the /v2/system-volumes endpoint
* FDE: support changing passphrase at runtime via the /v2/system-volumes endpoint
* FDE: support passphrase quality check at runtime via the /v2/system-volumes endpoint
* FDE: update secboot to revision 3e181c8edf0f

> * Confdb: support lists and indexed paths on write
> * Confdb: support list and indexed paths on read

Reviewer: Can these be combined into one? Or is `lists` vs `list` distinction important?

```suggestion
* Confdb: support lists and indexed paths on read and write
```

* Confdb: alias references must be wrapped in brackets
* Confdb: support indexed paths in confdb-schema assertion
* Confdb: make API errors consistent with options
* Confdb: fetch confdb-schema assertion on access
* Confdb: prevent `--previous` from being used in read-side hooks
* Components: fix snap command with multiple components
* Components: set revision of seed components to x1
* Components: unmount extra kernel-modules components mounts
* AppArmor Prompting: add lifespan "session" for prompting rules
* AppArmor Prompting: support restoring prompts after snapd restart

> * AppArmor Prompting: limit the extra information included in probed AppArmor features and system key

Reviewer: Actually, this isn't AppArmor Prompting-specific, this was more about addressing LP2116974. So perhaps this should be renamed and moved below?

```suggestion
* LP: #2116974 limit the extra information included in probed AppArmor features and system key
```

Though CPC has their own workaround, so technically this solution isn't the one which is fixing the problem, at least temporarily.

Author: I'm avoiding adding this to release notes, as its hard to prove what it fixes the problem.

> * Notices: refactor notice state internals

Author: I agree its internal, but its here on Oliver request.

> * Notices: refactor notice state internals

Reviewer: Yes I requested it because those changes are relevant to pebble as well they'll need to be ported there.

* SELinux: look for restorecon/matchpathcon at all known locations rather than current PATH
* SELinux: update policy to allow watching cgroups (for RAA), and talking to user session agents (service mgmt/refresh)
* Refresh App Awareness: Fix unexpected inotify file descriptor cleanup
* snap-confine: workaround for glibc fchmodat() fallback and handle ENOSYS
* snap-confine: add support for host policy for limiting users able to run snaps
* LP: #2114923 Reject system key mismatch advise when not yet seeded
* Use separate lanes for essential and non-essential snaps during seeding and allow non-essential installs to retry

> * Fix bug preventing remodel from core18 to core18 when snapd snap is unchanged

Reviewer: Is "core18 to core18" a typo or correct?
